File: Codes in Python3/Graph/Detect Cycle using DFS in Undirected.py
def isCycle(src,parent,adj,vis)->bool:
    vis[src]=True
    for adjacent in adj[src]:
        if vis[adjacent] is False:
            if isCycle(adjacent,src,adj,vis):
                return True
        elif parent!=adjacent:
            return True
    return False
def cycleDetection(edges, n, m):
    vis=[False]*(n+1)
    adj=[[] for i in range(n+1)]
    for i in range(m):
        adj[edges[i][0]].append(edges[i][1])
        adj[edges[i][1]].append(edges[i][0])
    for i in range(n+1):
        if vis[i] is False:
            if isCycle(i,-1,adj, vis):
                return "Yes"
    return "No"

File: Codes in C++/Graph/Detect Cycle using DFS in Undirected.cpp
bool isCycle(int src, int parent ,vector<int> adj[], vector<int> &vis)
{
    vis[src] = 1;
        for (auto adjacent : adj[src])
        {
            if (!vis[adjacent])
            {
                if(isCycle(adjacent,src,adj,vis))
                return true;
            }
            else if (parent != adjacent)
            {
                return true;
            }
        }
    
    return false;
}

string cycleDetection(vector<vector<int>> &edges, int n, int m)
{
    vector<int> adj[n + 1];
    for (int i = 0; i < m; i++)
    {
        int u = edges[i][0];
        int v = edges[i][1];
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    int v = n;
    vector<int> vis(v + 1, 0);
    for (int i = 0; i < v; i++)
    {
        if (!vis[i])
            if (isCycle(i, -1,adj, vis))
                return "Yes";
    }
    return "No";
}
